src/app/gene-plot/gene-plot.component.ts

```diff
@@ -200,7 +199,21 @@ export class GenePlotComponent implements OnChanges {
     this.redraw();
   }
 
+  public isInZoom(): boolean {
+    if (!((this.xDomain[1] === this.genePlotModel.domain[this.genePlotModel.domain.length - 1])
+      && (this.xDomain[0] === this.genePlotModel.domain[0]))) {
+        return true;
+    } else {
+      return false;
+    }
+  }
+
+  public map(x, in_min, in_max, out_min, out_max): number {
```

Comment:
d3 provides a `invert` method in the scale objects for this purpose

Reply:
additionally, this does not handle the case for zooming in on the denovo axis

Reply:
for selections overlapping the denovo axis, handle them by simply drawing either of the selection lines at the top or bottom of the axis, depending on how the zoom was made, e.g.: 
- entire zoom is within the denovo axis: draw borders on top and bottom of denovo axis
- only the lower bound of the zoom is within the denovo axis: draw the bottom border on the bottom of the denovo axis

Reply:
refer to `getAlleleHeight` for a reference on how to convert frequency values to Y coordinates